### DISCUSSION TITLE
How do I disable auto-resize for a window?

### DISCUSSION THREAD

**Discussion by kotx** [emilk/egui#610] [Q&A]
I want my window to be fixed-sized, but the user should be able to resize it manually. Ideally the fixed-size window should fit all widgets in it.

**Comment by coderedart** [comment]
Isn't that the default behaviour?
Just enable scrolling for window.

**Reply by kotx** [reply]
No, it's not- unless this https://github.com/emilk/egui/blob/9603bb4f85fbad328bce527ff0850fbc925393d6/egui_demo_lib/src/apps/demo/tests.rs#L424 is a bug. It might be.

**Comment by emilk** [comment]
Can you explain a bit more? "Fixed size" and "resizable" are opposites - you can only have one.

**Reply by kotx** [reply]
I mean that egui shouldn't resize it, but I want users to be able to resize it.

**Reply by emilk** [reply]
`.resizable(true)` should do it, IF the contents allow resizing. If the content is a single button, the window will still shrink to the size of that button.

Check out the "Window Resize" demo at https://emilk.github.io/egui/ - the source code is at https://github.com/emilk/egui/blob/master/egui_demo_lib/src/apps/demo/tests.rs#L386-L461

**Reply by kotx** [reply]
> `.resizable(true)` should do it, IF the contents allow resizing. If the content is a single button, the window will still shrink to the size of that button.
> 
> Check out the "Window Resize" demo at https://emilk.github.io/egui/ - the source code is at https://github.com/emilk/egui/blob/master/egui_demo_lib/src/apps/demo/tests.rs#L386-L461

Yeah but it seems I'm having the same issue as here https://github.com/emilk/egui/blob/9603bb4f85fbad328bce527ff0850fbc925393d6/egui_demo_lib/src/apps/demo/tests.rs#L424
where the window snaps sizes to the scroll view.

**Reply by emilk** [reply]
That's not a bug, but a consequence/limitation of immediate mode. The `ScrollArea` must be the last thing in the window for auto-resizing it to work, unless you use panels like in egui_demo_lib/src/apps/demo/window_with_panels.rs

**Reply by kotx** [reply]
So would it be possible to have the window be auto-sized initially, but never auto-size to the ScrollArea after?
